pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -145,24 +146,43 @@ def _configure_checkpoint_callbacks(self, checkpoint_callback: bool) -> None:
         if not self._trainer_has_checkpoint_callbacks() and checkpoint_callback is True:
             self.trainer.callbacks.append(ModelCheckpoint())
 
-    def _configure_model_summary_callback(self, weights_summary: Optional[str] = None) -> None:
+    def _configure_model_summary_callback(
+        self, enable_model_summary: bool, weights_summary: Optional[str] = None
+    ) -> None:
+        if not enable_model_summary:
+            return
         if any(isinstance(cb, ModelSummary) for cb in self.trainer.callbacks):
```

Comment:
what if there is a callback and weights_summary is not None? shouldn't we raise an exception for that case?

Reply:
I don't see a way we can do this now, as the existing default is `weights_summary="top"`
If we change the default value to `weights_summary=None`, then it's a breaking change, as users today use `weights_summary=None` to configure whether this is displayed or not. After this PR, they'd be forced to use `enable_model_summary=False` since there'd be no way to distinguish precedence/user input vs defaults between `enable_model_summary=True` and `weights_summary=None` 

So if we want to preserve the existing disable path, we can't change the default for `weights_summary`. 
On the other hand, if someone is using a custom model summary callback, they need all of `enable_model_summary=True`, `weights_summary != None` and the custom callback extending `ModelSummary`. and raising an exception here would block that use case entirely.

After `weights_summary` is deprecated, we can definitely strengthen this check